PGIP/Server.hs

```diff
@@ -1806,27 +1822,224 @@ consNode :: LibEnv -> LibName -> DGraph -> (Int, DGNodeLab)
+                          reasonerConfiguration goalConfig
+                    let caseReasoningCacheEntry = ReasoningCacheGoal
+                          { rceProverMode = proverMode
+                          , rceNode = node_
+                          , rceGoalNameM = Nothing
+                          , rceGoalConfig = goalConfig
+                          , rceGTheory = gTheory
+                          , rceGSublogic = gSublogic
+                          , rceReasoner = undefined -- will be overwritten a few lines below
+                          , rceComorphism = undefined -- will be overwritten a few lines below
+                          , rceTimeLimit = timeLimit_
+                          , rceUseDatabase = useDatabase
+                          , rceReasonerConfigurationKeyM = Nothing
+                          , rceReasoningAttemptKeyM = Nothing
+                          }
+                    case proverMode of
+                      GlConsistency -> do
+                        (gConsChecker, comorphism) <-
+                          findConsChecker translationM gSublogic reasonerM
+                        return ((Right $ caseReasoningCacheEntry
+                                 { rceReasoner = AbsState.ConsChecker gConsChecker
+                                 , rceComorphism = comorphism
+                                 }
+                               ) : reasoningCacheE2)
+                      GlProofs -> do
+                        proversAndComorphisms <-
+                          getProverAndComorph reasonerM translationM gSublogic
+                        (gProver, comorphism) <- case proversAndComorphisms of
+                          [] -> fail ("No matching translation or prover found for "
+                                      ++ nodeName ++ "\n")
+                          (gProver, comorphism) : _ ->
+                            return (gProver, comorphism)
+                        let possibleGoalNames = map fst $ getThGoals gTheory :: [String]
+                        let goalNames = (case conjecture goalConfig of
+                              Nothing -> possibleGoalNames
+                              Just goalName_ ->
+                                filter (== goalName_) possibleGoalNames) :: [String]
+                        return
+                          (map (\ goalName_ -> Right $ caseReasoningCacheEntry
+                                 { rceGoalNameM = Just goalName_
+                                 , rceReasoner = AbsState.Prover gProver
+                                 , rceComorphism = comorphism
+                                 }
+                               ) goalNames ++ reasoningCacheE2)
+                  )
+                  reasoningCacheE1
+                  goalConfigByNode
+      )
+      [] $
+      groupBy (\ a b -> ReasoningParameters.node a == ReasoningParameters.node b) $
+        ReasoningParameters.goals reasoningParameters
```

Comment:
I find it hard to review `buildReasoningCache`. Could you please give names to the anonymous functions (even if this means passing more arguments around), and also to the `groupby` term at the end?

Reply:
Will do. Could take me some time because I'm working on a different task right now.